components/prism-stylus.js

```diff
@@ -0,0 +1,61 @@
+Prism.languages.stylus = {
+	'comment': {
+		pattern: /(^|[^\\])(\/\*[\w\W]*?\*\/|\/\/.*?(\r?\n|$))/g,
+		lookbehind: true
+	},
+	'keyword': /(px|r?em)\b|\b(is|defined|not|isnt|and|or|unless|for|in)\b/g,
+	'atrule': /@[\w-]+(?=\s+\S+)/gi,
+	'url': /url\((["']?).*?\1\)/gi,
+	'variable': /^\s*([\w-]+)(?=\s*[+-\\]?=)/gm,
+	'string': /("|')(\\\n|\\?.)*?\1/g,
+	'important': /\B!important\b/gi,
+	'hexcode': /#[\da-f]{3,6}/gi,
+	'entity': /\\[\da-f]{1,8}/gi,
+	'number': /\d+\.?\d*%?/g,
+	'selector': [
+		{
+			pattern: /:{1,2}after|before|first-letter|first-line|selection/g,
+			alias: 'pseudo-element'
+		},{
+			pattern: /:(?:active|checked|disabled|empty|enabled|first-child|first-of-type|focus|hover|in-range|invalid|lang|last-child|last-of-type|link|not|nth-child|nth-last-child|nth-last-of-type|nth-of-type|only-of-type|only-child|optional|out-of-range|read-only|read-write|required|root|target|valid|visited)(?:\(.*\))?/g,
+			alias:'pseudo-class'
+		},{
+			pattern: /(\[[\w-]+?\s*[*~|$^]?(?:=\s*\S+)?\])/g,
```

Comment:
Why did you make the first pattern `[\w-]+` ungreedy? It can only be follow by an operator or a closing square bracket so it should be safe to keep it greedy.

What about selectors like `input[type]`? All the value-related part should be made optional, I think.

In the part for the value, `\S+` is greedy. Couldn't it be a problem with selectors like `div[foo=bar][baz=bar]`?

Finally, the parentheses around the whole regexp aren't needed.